var/spack/repos/builtin/packages/podio/package.py

```diff
@@ -72,12 +73,22 @@ class Podio(CMakePackage):
 
     depends_on("cxx", type="build")  # generated
 
+    _cxxstd_values = (conditional("17", when="@:1.2"), conditional("20", when="@0.15:"))
     variant(
         "cxxstd",
         default="17",
-        values=("17", conditional("20", when="@0.15:")),
+        values=_cxxstd_values,
         multi=False,
         description="Use the specified C++ standard when building.",
+        when="@:1.1",
+    )
+    variant(
+        "cxxstd",
+        default="20",
+        values=_cxxstd_values,
+        multi=False,
+        description="Use the specified C++ standard when building.",
+        when="@1.2:",
```

Comment:
Yes, duplication is for default value only. I don't want to change the default on older version (in part because a default 17 would not even be allowed for older versions). There is no conditional for default values (or wasn't last I checked).

Reply:
Makes sense. In principle all versions still available here already can be built with c++20, so purely from that point of view changing the default shouldn't break ~anything~too much. I am OK with keeping this as it is for now.

Reply:
I just don't want to break someone's carefully constructed environment by suddenly switching a default on an older version :-)